tests/test_preload_script.py

```diff
@@ -74,7 +95,280 @@ async def test_addPreloadScript_logging(websocket, context_id, html):
+                    iframe.src = '{url_cross_origin}';
+                    document.body.appendChild(iframe);""",
+                "target": {
+                    "context": context_id
+                },
+                "awaitPromise": True,
+                "resultOwnership": "root"
+            }
+        })
+
+    # Asset that the preload script is executed in the new iframe.
+    result = await read_JSON_message(websocket)
+    assert result == AnyExtending({
+        "method": "log.entryAdded",
+        "params": {
+            "args": [{
+                "type": "string",
+                "value": "my preload script"
+            }]
+        }
+    })
+
+
+@pytest.mark.asyncio
+async def test_addPreloadScript_loadedInNewIframes_withSameOrigin(
+        websocket, context_id, url_same_origin, html):
+    await subscribe(websocket, "log.entryAdded")
+
+    await execute_command(
+        websocket, {
+            "method": "script.addPreloadScript",
+            "params": {
+                "expression": "() => console.log('my preload script')",
+                "context": context_id,
+            }
+        })
+
+    command_id = await send_JSON_command(
+        websocket, {
+            "method": "browsingContext.navigate",
+            "params": {
+                "url": html(),
+                "wait": "complete",
+                "context": context_id
+            }
+        })
+
+    # Log event should happen before navigation.
+    result = await read_JSON_message(websocket)
+    assert result == AnyExtending({
+        "method": "log.entryAdded",
+        "params": {
+            "args": [{
+                "type": "string",
+                "value": "my preload script"
+            }]
+        }
+    })
+
+    # Assert navigation is finished.
+    result = await read_JSON_message(websocket)
+    assert result == {"id": command_id, "result": ANY_DICT}
+
+    # Create a new iframe within the same context.
+    result = await send_JSON_command(
+        websocket, {
+            "method": "script.evaluate",
+            "params": {
+                "expression": f"""const iframe = document.createElement('iframe');
+                    iframe.src = '{url_same_origin}';
+                    document.body.appendChild(iframe);""",
+                "target": {
+                    "context": context_id
+                },
+                "awaitPromise": True,
+                "resultOwnership": "root"
+            }
+        })
+
+    # Asset that the preload script is executed in the new iframe.
+    result = await read_JSON_message(websocket)
+    assert result == AnyExtending({
+        "method": "log.entryAdded",
+        "params": {
+            "args": [{
+                "type": "string",
+                "value": "my preload script"
+            }]
+        }
+    })
+
+    result = await read_JSON_message(websocket)
+
+
+@pytest.mark.asyncio
+async def test_addPreloadScript_loadedInNewIframes_withChildScript(
```

Comment:
Is it any different from the previous 2 tests?

Reply:
The `iframe` with `console.log` is the differential. Do you want me to delete it? Deleting is easier than writing ;)

Reply:
Oh, I see. Then it should be called smth like `test__addPreloadScript_runBeforePageScript`.